launcher/agent/agent_test.go

```diff
@@ -255,3 +256,82 @@ func TestFetchContainerImageSignatures(t *testing.T) {
 		})
 	}
 }
+
+func TestCacheConcurrentGet(t *testing.T) {
+	ctx := context.Background()
+
+	cache := &sigsCache{}
+	if sigs := cache.get(); len(sigs) != 0 {
+		t.Errorf("signature cache should be empty, but got: %v", sigs)
+	}
+
+	sdClient := signaturediscovery.NewFakeClient()
+	targetRepos := []string{signaturediscovery.FakeRepoWithSignatures}
+	original := fetchContainerImageSignatures(ctx, sdClient, targetRepos, log.Default())
+	cache.set(original)
+
+	wantBase64Sigs := convertOCISignatureToBase64(t, original)
+
+	group, _ := errgroup.WithContext(ctx)
+	for i := 0; i < runtime.NumCPU(); i++ {
+		group.Go(func() error {
+			cached := cache.get()
+			gotBase64Sigs := convertOCISignatureToBase64(t, cached)
+			if !cmp.Equal(wantBase64Sigs, gotBase64Sigs) {
+				return fmt.Errorf("cached base64 sigs not equal to original base64 sigs, got %v, but want %v", gotBase64Sigs, wantBase64Sigs)
+			}
+			return nil
+		})
+	}
+
+	if err := group.Wait(); err != nil {
+		t.Errorf("cache concurrent get failed: %v", err)
+	}
+}
+
+func TestCacheConcurrentSet(t *testing.T) {
+	ctx := context.Background()
+
+	cache := &sigsCache{}
+	if sigs := cache.get(); len(sigs) != 0 {
+		t.Errorf("signature cache should be empty, but got: %v", sigs)
+	}
+
+	sdClient := signaturediscovery.NewFakeClient()
+	targetRepos := []string{signaturediscovery.FakeRepoWithSignatures}
+
+	var wg sync.WaitGroup
+	for i := 0; i < runtime.NumCPU(); i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			sigs := fetchContainerImageSignatures(ctx, sdClient, targetRepos, log.Default())
+			cache.set(sigs)
```

Comment:
Since we aren't intermingling sets and gets in a concurrent manner, we aren't actually testing the sync primitive. Consider intermingling data accesses and using the `-race` flag: https://go.dev/doc/articles/race_detector

Reply:
Done.